cadctl/cmd/credentials/credentials.go

```diff
@@ -0,0 +1,73 @@
+package credentials
+
+import (
+	"fmt"
+	"github.com/openshift/configuration-anomaly-detection/cadctl/cmd/investigate"
+	"github.com/openshift/configuration-anomaly-detection/pkg/services/assumerole"
+	"github.com/openshift/configuration-anomaly-detection/pkg/services/ccam"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+var Cmd = &cobra.Command{
+	Use:   "credentials",
+	Short: "Determines if a cluster's credentials are missing.",
+	Long: `Determines if a cluster's credentials are missing. If the command determines that they are,
+the cluster is placed into limited support. If the cluster is already in limited support, the command
+will evaluate if it the credentials have been restored and remove it from limited support if appropriate.`,
+	RunE: run,
+}
+
+var clusterId string
+
+func init() {
+	Cmd.Flags().StringVarP(&clusterId, "cluster-id", "c", "", "OCM internal/external cluster id to run against.")
+	_ = Cmd.MarkFlagRequired("cluster-id")
+}
+
+func run(_ *cobra.Command, _ []string) error {
+	awsClient, err := investigate.GetAWSClient()
+	if err != nil {
+		return fmt.Errorf("could not initialize aws client: %w", err)
+	}
+
+	ocmClient, err := investigate.GetOCMClient()
+	if err != nil {
+		return fmt.Errorf("could not initialize ocm client: %w", err)
+	}
+
+	arClient := assumerole.Client{
+		Service: assumerole.Provider{
+			AwsClient: awsClient,
+			OcmClient: ocmClient,
+		},
+	}
+
+	cssJumprole, ok := os.LookupEnv("CAD_AWS_CSS_JUMPROLE")
+	if !ok {
+		return fmt.Errorf("CAD_AWS_CSS_JUMPROLE is missing")
+	}
+
+	supportRole, ok := os.LookupEnv("CAD_AWS_SUPPORT_JUMPROLE")
+	if !ok {
+		return fmt.Errorf("CAD_AWS_SUPPORT_JUMPROLE is missing")
+	}
+
+	cluster, err := ocmClient.GetClusterInfo(clusterId)
+	if err != nil {
+		return fmt.Errorf("could not retrieve cluster info for %s: %w", clusterId, err)
+	}
+
+	ccamClient := ccam.Client{
```

Comment:
**issue:** This won't work, as the `Evaluate` function has API calls to pagerduty.